### PR TITLE
Hibernate jsonb support

### PR DESCRIPTION
**Pull request by pratickchokhani**

### CODE REVIEW

File: samples/java/hibernate/README.md [hunk 1]
```diff
@@ -50,14 +50,22 @@ Cloud Spanner supports the following data types in combination with `Hibernate`.
 | timestamptz / timestamp with time zone | LocalDateTime     |
 | bytea                                  | byte[]            |
 | date                                   | LocalDate         |
+| jsonb                                  | Custom Data Type  |
+
+### JSONB
+Hibernate 5.* does not have native support for JSONB. For this, custom data type
+will have to be added. There are modules available that provide support for this.
+For the sample, we have used [vladmihalcea/hibernate-types](https://github.com/vladmihalcea/hibernate-types).
```
Comment (olavloite) [inline samples/java/hibernate/README.md]:
```suggestion
Hibernate 5.* does not have native support for JSONB. For this, a custom data type
must be added. This sample uses [vladmihalcea/hibernate-types](https://github.com/vladmihalcea/hibernate-types).
```

File: samples/java/hibernate/README.md [hunk 2]
```diff
@@ -50,14 +50,22 @@ Cloud Spanner supports the following data types in combination with `Hibernate`.
...
+| Hibernate Version      | Version 5.3.20.Final is supported.                                                                                                                                                                                                                                           |
+| Postgresql JDBC driver | Version 42.4.3 is supported.                                                                                                                                                                                                                                                 |
+| Schema updates         | Cloud Spanner does not support the full PostgreSQL DDL dialect. Automated schema updates using `hibernate` are therefore not supported. It is recommended to set the option `hibernate.hbm2ddl.auto=none` (or `spring.jpa.hibernate.ddl-auto=none` if you are using Spring). |
+| Generated primary keys | Cloud Spanner does not support `sequences`. Auto-increment primary key is not supported. Remove auto increment annotation for primary key columns. The recommended type of primary key is a client side generated `UUID` stored as a string.                                 |
```
Comment (olavloite) [inline samples/java/hibernate/README.md:67]: I think we should add pessimistic locking to this table as well (https://docs.jboss.org/hibernate/stable/core.old/reference/en/html/transactions-locking.html). `SELECT ... FOR UPDATE` is currently not supported by Cloud Spanner, and although there are many other backend features that are also not supported, this is one that is used more often by Hibernate users and therefore deserves a specific mention.

**Review by olavloite — APPROVED** [review]
LGTM (with a couple of nits)

File: samples/java/hibernate/README.md (hunk 2 above)
Comment (pratickchokhani) [reply samples/java/hibernate/README.md:67]: Done

**Pull request merged**